File: 71-simplify-path/simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    const splitPath = path.split("/");

    const simplifiedPath = []
    for (let el of splitPath) {
        if (el && !(el == ".") & !(el == "..")) {
            simplifiedPath.push(el)
        }
        if (el == "..") {
            simplifiedPath.pop()
        }
    }
    return `/${simplifiedPath.join("/")}`
};

File: 71-simplify-path/simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        splitPath = path.split("/")

        stack = []

        for ele in splitPath:
            if ele and ele not in ("..", "."):
                stack.append(ele)
            if (ele == ".." and stack):
                stack.pop()

        return f"/{'/'.join(stack)}"